### PR TITLE
Adding interface for inria museum

### PR DESCRIPTION
**Pull request by ChloeMarieRiviere**

### CODE REVIEW

File: InriaMuseumInterface/.ipynb_checkpoints/base-checkpoint.ipynb
Comment (twatteyne) [inline InriaMuseumInterface/.ipynb_checkpoints/base-checkpoint.ipynb:1]: add `*.ipynb` files to `.gitignore`

File: InriaMuseumInterface/1.svg
Comment (twatteyne) [inline InriaMuseumInterface/1.svg:1]: give these files a name, not just a number

File: InriaMuseumInterface/base.ipynb
Comment (twatteyne) [inline InriaMuseumInterface/base.ipynb:1]: add `*.ipynb` to `.gitignore`

File: InriaMuseumInterface/__pycache__/bottle.cpython-311.pyc
Comment (twatteyne) [inline InriaMuseumInterface/__pycache__/bottle.cpython-311.pyc:1]: add `__pycache__` to .gitignore

File: InriaMuseumInterface/demoMusee.css
Comment (twatteyne) [inline InriaMuseumInterface/demoMusee.css:1]: rename all files/functions related to the Inria Museum as `InriaMusem`; be very very consistent.

File: InriaMuseumInterface/static/demoMusee.css
Comment (twatteyne) [inline InriaMuseumInterface/static/demoMusee.css:1]: why two `css`?

**Review by twatteyne — CHANGES_REQUESTED** [review]
Great first PR. My main comment is that you need to clean up what you commit, the naming, etc. Please see individual comments. Commit to the same branch, chat in GitHub if you need, and let me know when you want me to re-review.

File: InriaMuseum/svg/Activate_mote.svg
Comment (twatteyne) [inline InriaMuseum/svg/Activate_mote.svg:1]: all file names lowercase please

File: InriaMuseum/svg/Welcom.svg
Comment (twatteyne) [inline InriaMuseum/svg/Welcom.svg:1]: welcome

File: InriaMuseum/__pycache__/bottle.cpython-311.pyc
Comment (twatteyne) [inline InriaMuseum/__pycache__/bottle.cpython-311.pyc:1]: remove this file/folder from GitHub

File: InriaMuseum/InriaMuseum.html [hunk 1]
```diff
@@ -0,0 +1,153 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <title>Musé Inria exposition</title>
```
Comment (twatteyne) [inline InriaMuseum/InriaMuseum.html]: Inria Museum

File: InriaMuseum/InriaMuseum.html [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+    <link rel="stylesheet" type="text/css" href="InriaMuseum.css"> 
+    <script>
+
+        var current_page;
+        function handleButtonClick(event) {
+            if(event.target.id == ""){
+                const parentElementId = event.target.parentElement ? event.target.parentElement.id : 'No parent element';
+                changePage(parentElementId);
+            }
+            else{
```
Comment (twatteyne) [inline InriaMuseum/InriaMuseum.html]: `} else {`

File: InriaMuseum/InriaMuseum.html [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+                changePage(event.target.id);
+
```
Comment (twatteyne) [inline InriaMuseum/InriaMuseum.html]: remove newline

File: InriaMuseum/InriaMuseum.html [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                case "button_défis":
+                    current_page='Select_music.svg';
+                    break;
+                default :
+                    console.log(`default ${ElementId} `); 
+                    break;
+            }
+            loadSVG(current_page);
+            checkPage();
+
```
Comment (twatteyne) [inline InriaMuseum/InriaMuseum.html]: remove mnewline

File: InriaMuseum/InriaMuseum.html [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+                    current_page='Activate_mote.svg';
+                    break;
+                case "Activate_mote.svg":
+                    current_page='Listening.svg';
+                    break;
+                case "Listening.svg":
+                    current_page='Ending_Demonstration.svg';
+                    break;
+            }               
+
```
Comment (twatteyne) [inline InriaMuseum/InriaMuseum.html]: remove newline

File: InriaMuseum/InriaMuseum.html [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+        }
+
+        function checkPage() {
+            if(current_page=="Launching.svg"){
+                current_page = "Introduction.svg";
+                setTimeout(function(){
```
Comment (twatteyne) [inline InriaMuseum/InriaMuseum.html:72]: fix indentation

File: InriaMuseum/InriaMuseum.html [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+    loadSVG(current_page);
+}, 5000);
+            }
+            if(current_page=="Listening.svg"){
+                current_page = "Ending_Demonstration.svg";
+                setTimeout(function(){
+    loadSVG(current_page);
+}, 5000);
+
```
Comment (twatteyne) [inline InriaMuseum/InriaMuseum.html]: remove newline

File: InriaMuseum/InriaMuseum.html [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+            }
+            if(current_page=="Listening.svg"){
+                current_page = "Ending_Demonstration.svg";
+                setTimeout(function(){
+    loadSVG(current_page);
+}, 5000);
+               
+            }
+        }
+
```
Comment (twatteyne) [inline InriaMuseum/InriaMuseum.html:93]: just one empty line

File: InriaMuseum/InriaMuseum.html [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+        function loadSVG(filename) {
+            const filePath = `svg/${filename}`;
+            console.log(`New page ${filename} `);
+            fetch(filePath)
+                .then(response => response.text())
+                .then(svgContent => {
+                    document.getElementById('svgContainer').innerHTML = svgContent;
+                    loadelement();
+
```
Comment (twatteyne) [inline InriaMuseum/InriaMuseum.html]: remove newline

File: InriaMuseum/InriaMuseum.html [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+                })
+                .catch(error => console.error('Error loading SVG:', error));
+        }
+        
+        window.onload = function() {
+            loadSVG('Welcom.svg');
+            loadelement();
+        }
+
```
Comment (twatteyne) [inline InriaMuseum/InriaMuseum.html:110]: only one empty line

File: InriaMuseum/InriaMuseum.html [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+                const element = document.getElementById(id);
+                if (element) {
+                    element.addEventListener("click", handleButtonClick);
+                    console.log(`Event listener attached to: ${id}`);
+                } else {
+                    console.log(`Element with id ${id} not found`);
+                }
+            });
+
+              if(current_page=="Describe_network.svg"){
```
Comment (twatteyne) [inline InriaMuseum/InriaMuseum.html]: fix indentation

File: InriaMuseum/InriaMuseum.html [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+                changeColor("blue","xplore_mote");
+                changeColor("orange","algorithme_board");
+            }
+
```
Comment (twatteyne) [inline InriaMuseum/InriaMuseum.html]: remove both newlines

File: InriaMuseum/bottle.py
Comment (twatteyne) [inline InriaMuseum/bottle.py:1]: don't add this file here, just do a `pip install bottle`

**Review by twatteyne — CHANGES_REQUESTED** [review]
more small changes, please add a `README.md` file indicating how to start the demo. I dont see a Python file.

File: InriaMuseum/README.md [hunk 13]
```diff
@@ -0,0 +1,6 @@
+# Inria Museum
+
+This part is destined to the Inria museum for the exposiotion. 
+In this part you will find a web interface to launch on the Raspberry Pi.
+
+This interface is coded using : Python, bottle, HTML, SVG & Javascript
```
Comment (twatteyne) [inline InriaMuseum/README.md]: describe how ot run it

File: InriaMuseum/requirements.TXT [hunk 14]
```diff
@@ -0,0 +1 @@
+bottle
```
Comment (twatteyne) [inline InriaMuseum/requirements.TXT]: rename txt, not TXT

File: InriaMuseum/InriaMuseum.css [hunk 15]
```diff
@@ -0,0 +1,7 @@
+#button_suite_group,#button_start_fr,#button_active,#button_leave,#button_music1,#button_music2,#button_défis,#button_défis-2:hover{
+    cursor:grab;
+}
+
+.doc:hover{
+    cursor:default;
+}
```
Comment (twatteyne) [inline InriaMuseum/InriaMuseum.css]: add an empty newline

File: InriaMuseum/InriaMuseum.css [hunk 16]
```diff
@@ -0,0 +1,7 @@
+#button_suite_group,#button_start_fr,#button_active,#button_leave,#button_music1,#button_music2,#button_défis,#button_défis-2:hover{
```
Comment (twatteyne) [inline InriaMuseum/InriaMuseum.css:1]: why all of these classes, please remove everything that is not used

File: InriaMuseum/InriaMuseum.py [hunk 17]
```diff
@@ -0,0 +1,11 @@
...
+
+@route('/')
+def index():
+    return static_file('InriaMuseum.html', root='.')
+
+@route('/<filename:path>')
+def serve_static(filename):
+    return static_file(filename, root='.')
+
+run(host='localhost', port=8080)
```
Comment (twatteyne) [inline InriaMuseum/InriaMuseum.py]: add empty line

**Review by twatteyne — CHANGES_REQUESTED** [review]
further suggestions, please address all comments

File: InriaMuseum/InriaMuseum.css (hunk 16 above)
Comment (ChloeMarieRiviere) [reply InriaMuseum/InriaMuseum.css:1]: All of them are used is the id of the button to transform the mouse in an hand

File: .gitignore [hunk 18]
```diff
@@ -0,0 +1,7 @@
+# Avoid directory
+**/__pycache__/
```
Comment (aabadie) [inline .gitignore:2]: You already ignore "*.pyc" pattern so it's not really needed to ignore the `__pycache__` directory.

File: .gitignore [hunk 19]
```diff
@@ -0,0 +1,7 @@
...
+
+# Avoid file
+*.ipynb
+**/bottle.py
+*.pyc
```
Comment (aabadie) [inline .gitignore:7]: There's a missing newline at the end of this file.

File: InriaMuseum/InriaMuseum.py [hunk 20]
```diff
@@ -0,0 +1,58 @@
...
+    try:
+        response = requests.get(SERVERtoLISTEN)
+        if response.status_code == 200:
+            json_data = response.json()  
+            print(f"Message receive : {json_data}")
+            json= json_data.get("payload")
+            data = Get_position(json)
+            return data
+        else:
+            print("Error : recieve")
```
Comment (aabadie) [inline InriaMuseum/InriaMuseum.py:42]:
```suggestion
            print("Error : receive")
```

**Review by aabadie — COMMENTED** [review]
There are typos in the svg files:
- ending_demonstration.svg: "utilitée" => "utilité"
- select_music.svg: "Quelle morceau" => "Quel morceau"

It's also a good practice to not use accentuated characters in css IDs ("button_défis" and "button_défis-2" should rather be called "button_challenge" and "button_challenge_2" IMO).

Other tiny comments inline.

**Comment by twatteyne** [comment]
These changes appear in #8 , so closing this one as duplicate

**Pull request closed without merging**